### DISCUSSION TITLE
How to pass OpenGL buffer into kernel? Or how to make device pointers (int type) into DeviceAllocation?

### DISCUSSION THREAD

**Discussion by zongyuelu** [inducer/pycuda#347] [Q&A]
Currently I got a OpenGL buffer object, and I did this as in docs
```python
from pycuda import gl as cudaGL
from pycuda.compiler import SourceModule

registeredBuffer = cudaGL.RegisteredBuffer(int(glBuffer))
registeredMapping = registeredBuffer.map()
devptr, size = registeredMapping.dev_ptr_and_size()

srcCode = """
__global__ void foo(float* src) { ... }
"""
func = SourceModule(srcCode).get_function('foo')
func(devptr, block=(32, 32, 1)) # no, devptr can't be recognized as C pointer
```
where `devptr, size` here are `int` type, **how to convert these into a `DeviceAllocation` so that it can be passed into a kernel?** Since doc says `Function.__call__` method only takes `numpy.number` and `DeviceAllocation` as inputs.

**Comment by inducer** [comment]
Use a sized integer (`np.intp`) or prepare the call with the appropriate type signature (`P`).
